subsys/pm/device.c

```diff
@@ -28,9 +43,8 @@ const char *pm_device_state_str(enum pm_device_state state)
 int pm_device_action_run(const struct device *dev,
```

Comment:
I think we should avoid adding more logic to `pm_device_action_run`, as it encourages its usage over the runtime API, which I think should be preferred.

Reply:
The only logic being added to `pm_device_action_run` is setting `PM_DEVICE_FLAG_TURN_ON_FAILED` if `PM_ACTION_TURN_ON` fails, and clearing it when `PM_ACTION_TURN_OFF` runs. This seems like the correct location for this logic.